### PR TITLE
tracing: wrap EnvFilter layer in reloadable layer

### PR DESCRIPTION
**Pull request by bmwill**
Also fix our misuse of layering various optional tracing layers.

### CODE REVIEW

**Review by velvia — APPROVED** [review]
Looks awesome, thanks for doing this!

File: crates/telemetry-subscribers/tests/reload.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+#[test]
+fn reload() {
+    let (_guard, reload_handle) = TelemetryConfig::new("test").with_env().init();
+
+    info!("Should be able to see this");
+    debug!("This won't be captured");
+    reload_handle.reload(EnvFilter::new("debug")).unwrap();
+    debug!("Now you can see this!");
+}
```
Comment (huitseeker) [inline crates/telemetry-subscribers/tests/reload.rs]: Do you want to  [tracing_test](https://docs.rs/tracing-test/latest/tracing_test/) this?

**Review by huitseeker — APPROVED** [review]

File: crates/telemetry-subscribers/tests/reload.rs (hunk 1 above)
Comment (bmwill) [reply crates/telemetry-subscribers/tests/reload.rs]: I dislike tracing_test, but i'll do you one better and just use our infra to write to a file and check the contents of it.

Comment (huitseeker) [reply crates/telemetry-subscribers/tests/reload.rs]: Nice 😁

**Pull request merged**